feed.js

```diff
@@ -172,6 +183,17 @@ Feed.prototype._onmessage = function (type, data, start, end) {
 }
 
 Feed.prototype._emit = function (type, message) {
+  switch (type) {
```

Comment:
can you guard this switch with a debug.enabled or similar? this function is extremely hot